generator/parser/binder.cpp

```diff
@@ -292,6 +292,19 @@ void Binder::declare_symbol(SimpleDeclarationAST *node, InitDeclaratorAST *init_
       applyStorageSpecifiers(node->storage_specifiers, fun.staticCast<_MemberModelItem>());
       applyFunctionSpecifiers(node->function_specifiers, fun);
       
+      if (hasFriendSpecifier(node->storage_specifiers))
+      {
+        // check if this function declaration is a "friend" function.
+        // In this case we modify the scope, and remove the "friend" flag.
+        // This is for functions/operators that are only defined as friend
+        // inside of classes.
+        symbolScope = _M_current_file.staticCast<_ScopeModelItem>();
+        // unset the friend flag, as we treat this like a stand-alone function definition
+        fun->setFriend(false);
+        // also set the access policy to public, just in case
+        fun->setAccessPolicy(CodeModel::Public);
```

Comment:
Just in what case? Is this really needed?

Reply:
I simply copied this part from Binder::visitFunctionDefinition, where the same thing was done.